pkg/kv/kvserver/protectedts/ptreconcile/reconciler.go

```diff
@@ -135,12 +142,21 @@ func (r *Reconciler) reconcile(ctx context.Context) {
 			continue
 		}
 		var didRemove bool
+		var didExpire bool
+
 		if err := r.db.Txn(ctx, func(ctx context.Context, txn isql.Txn) (err error) {
 			didRemove = false // reset for retries
+			didExpire = false
 			shouldRemove, err := task(ctx, txn, rec.Meta)
 			if err != nil {
 				return err
 			}
+			if !shouldRemove && rec.Target != nil &&
+				rec.Target.Expiration > 0 &&
+				rec.Timestamp.Add(int64(rec.Target.Expiration), 0).Less(r.clock.Now()) {
```

Comment:
do you think this is an interesting enough event that we should log something here? Might be useful when we're looking at debug zips in the future

Reply:
+1

I'd say we should _at least_ warn here, since the _expectation_ is that this shouldn't happen: the expectation with PTS is that the owner chooses, explicitly, when to release it.

I'm still a little unsure if this is even the layer where we should be doing expirations at all, but I'll make a top-level comment on this instead.